imap_processing/ultra/l1b/cullingmask.py

```diff
@@ -25,14 +25,32 @@ def calculate_cullingmask(
     cullingmask_dataset : xarray.Dataset
         Dataset containing the data.
     """
-    cullingmask_dict = {}
-    cullingmask_dict["spin_number"] = extendedspin_dataset["spin_number"]
-    cullingmask_dict["energy_bin_geometric_mean"] = extendedspin_dataset[
-        "energy_bin_geometric_mean"
-    ]
-
-    # TODO: add more data to cullingmask_dict.
-
-    cullingmask_dataset = create_dataset(cullingmask_dict, name, "l1b", data_version)
+    # If the spin rate was too high or low then the spin should be thrown out.
+    rate_mask = (
+        extendedspin_dataset["quality_attitude"] & ImapAttitudeUltraFlags.SPINRATE.value
+    ) == 0
+    good_attitude = extendedspin_dataset["spin_number"].values[rate_mask]
+    good_attitude_dataset = extendedspin_dataset.sel(spin_number=good_attitude)
```

Comment:
I think you might be able to index into this directly without needing the intermediate. The `good_attitude` is confusing because it is actually `good_spins`.

```suggestion
    good_attitude_dataset = extendedspin_dataset.where(rate_mask, drop=True)
```